docs/modules/vector-search/pages/vector-collections.adoc

```diff
@@ -0,0 +1,498 @@
+                .setName("index-2")
+                .setDimension(2)
+                .setMetric(Metric.DOT)
+                .setMaxDegree(10)
+                .setEfConstruction(20)
+                .setUseDeduplication(false)
+    );
+config.addVectorCollectionConfig(collectionConfig);
+----
+--
+====
+
+== Manage collection
+
+=== Create collection
+
+To create a vector collection, use the static method `getCollection(HazelcastInstance instance, VectorCollectionConfig collectionConfig)` of the `VectorCollection` class.
+
+Example:
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, vectorCollectionConfig);
+----
+--
+====
+
+=== Get existing vector collection
+
+To get a vector collection, use the static method `getCollection(HazelcastInstance instance, String collectionName)` of the `VectorCollection` class.
+
+Example:
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+----
+--
+====
+
+=== Optimize collection
+
+An optimization operation may be needed in the following cases:
+
+* To permanently delete nodes from the graph that were marked for removal.
+* After adding a significant number of vectors.
+* If some nodes become unreachable and the search results return fewer vectors than expected.
+
+NOTE: The optimization operation is time-consuming process, and no mutable operations are allowed during this process.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+CompletionStage<Void> result = vectorCollection.optimizeAsync("index_name");
+----
+--
+====
+
+== Manage data
+All methods of `VectorCollection` that work with collection data are asynchronous. The result is returned as a `CompletionStage`. A collection interacts with objects in the form of documents (`VectorDocument`). Each document comprises a value and one or more vectors associated with that value.
+
+=== Create document
+To create a document use statics the methods of `VectorDocument` and `VectorValues` classes.
+
+Example single index vector collection:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorDocument<String> document = VectorDocument.of(
+        "document value",
+        VectorValues.of(
+                new float[]{0.2f, 0.9f, -1.2f}
+        )
+);
+----
+--
+====
+
+For multi-index collections, specify the names of the indexes to which the vectors belong:
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorDocument<String> document = VectorDocument.of(
+        "some document value",
+        VectorValues.of(
+                Map.of(
+                        "index1", new float[] {1.2f, -0.3f, 2.2f},
+                        "index2", new float[] {0.2f, 1.3f, -0.2f}
+                )
+        )
+);
+----
+--
+====
+
+
+=== Put objects
```

Comment:
are we calling them "objects" or "entries"? I prefer entries. But we use also VectorDocument and value 

entry = key + VectorDocument

Reply:
b2fd6c714396ed311c966efd372253f1d485ff18